frontend/src/hooks/useContainerDimensions.js

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2021 Dominic Heil <d.heil@campus.tu-berlin.de>
+
+import { useEffect, useState } from "react";
+
+// https://stackoverflow.com/questions/43817118/how-to-get-the-width-of-a-react-element
+/**
+ * Hook to get width and height of a ref.
+ * @param myRef
+ * @returns {{width: number, height: number}}
+ */
+export const useContainerDimensions = myRef => {
+	const getDimensions = () => ({
+		width: myRef.current.offsetWidth,
+		height: myRef.current.offsetHeight
+	});
+
+	const [dimensions, setDimensions] = useState({ width: 0, height: 0 });
+
+	useEffect(() => {
+		const handleResize = () => {
+			setDimensions(getDimensions());
+		};
+
+		if (myRef.current) {
+			setDimensions(getDimensions());
+		}
+
+		window.addEventListener("resize", handleResize);
+
+		return () => {
+			window.removeEventListener("resize", handleResize);
+		};
+	}, [myRef]);
```

Comment:
Hm, I don't know about this. When does `myRef` trigger this effect? Does this only happen, when `myRef` is created or replaced with a new reference value?

Reply:
Whenever `myRef ` changes and once at the start.

Reply:
Okay. Sorry for asking, so it sounds like, if the reference address changes but not the referenced content (?)

Reply:
https://medium.com/@teh_builder/ref-objects-inside-useeffect-hooks-eb7c15198780 explains this better then I could.